TES3Merge/Util/Installation.cs

```diff
@@ -381,58 +383,193 @@ public override string GetDefaultOutputDirectory()
-            return Path.Combine(home, "Library", "Preferences", "openmw", "openmw.cfg");
+            return Path.Combine(home, "Library", "Preferences", "openmw");
         }
 
         throw new Exception("Could not determine configuration path.");
     }
 
-    private void LoadConfiguration()
+    private static string GetDefaultUserDataDirectory()
     {
-        var configPath = GetConfigurationLocation();
+        if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+        {
+            var myDocs = Environment.GetFolderPath(Environment.SpecialFolder.MyDocuments);
+            return Path.Combine(myDocs, "My Games", "OpenMW");
+        }
+        else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+        {
+            var dataHome = Environment.GetEnvironmentVariable("XDG_DATA_HOME");
+
+            if (string.IsNullOrEmpty(dataHome))
+                dataHome = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.UserProfile), ".local", "share");
+
+            return Path.Combine(dataHome, "openmw");
+        }
+        else if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+        {
+            var home = Environment.GetFolderPath(Environment.SpecialFolder.UserProfile);
+            return Path.Combine(home, "Library", "Application Support", "openmw");
+        }
+
+        throw new Exception("Could not determine user data directory.");
+    }
+
+    private static string UnescapeAmpersands(string input)
+    {
+        var result = new System.Text.StringBuilder(input.Length);
+        for (int i = 0; i < input.Length; i++)
+        {
+            if (input[i] == '&')
+            {
+                // If next char is also &, keep one and skip the next
+                if (i + 1 < input.Length && input[i + 1] == '&')
+                {
+                    result.Append('&');
+                    i++;
+                }
+            }
+            else
+            {
+                result.Append(input[i]);
+            }
+        }
+        return result.ToString();
+    }
+
+    private static string ParseDataDirectory(string configDir, string dataDir)
+    {
+        if (dataDir.StartsWith('"'))
+        {
+            int lastQuote = -1;
+            for (int i = dataDir.Length - 1; i > 0; i--)
+            {
+                if (dataDir[i] == '"')
+                {
+                    int ampCount = 0;
+                    int j = i - 1;
+                    while (j >= 0 && dataDir[j] == '&')
+                    {
+                        ampCount++;
+                        j--;
+                    }
+                    if (ampCount % 2 == 0)
+                    {
+                        lastQuote = i;
+                        break;
+                    }
+                }
+            }
+            if (lastQuote > 0)
+            {
+                dataDir = dataDir.Substring(0, lastQuote + 1);
+            }
+            dataDir = UnescapeAmpersands(dataDir.Trim('"'));
+        }
```

Comment:
This looks *way* more complicated than it needs to be. It could be as simple as
```cs
if (dataDir.StartsWith('"'))
{
    auto original = dataDir;
    dataDir = "";
    for (int i = 1; i < Original.Length; i++)
    {
        if (Original[i] == '&')
            i++;
        else if (Original[i] == '"')
            break;
        dataDir += Original[i];
    }
}
}